src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -121,15 +121,56 @@ bool DefaultOTARequestorDriver::ProviderLocationsEqual(const ProviderLocationTyp
     }
 }
 
-void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error) {}
+void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error)
+{
+    if (error == CHIP_NO_ERROR)
+    {
+        return;
+    }
+
+    switch (state)
+    {
+    case UpdateFailureState::kUnknown:
+        // Retry with next provider in the list
+        ScheduleRetry(false);
+        break;
+    case UpdateFailureState::kIdle:
```

Comment:
Should we just remove this failure state from the enum if it's not being used and shouldn't be anyway?

Reply:
Removed UpdateFailureState::kIdle case.